src/main.ts

```diff
@@ -388,95 +433,76 @@ export default class ReadwisePlugin extends Plugin {
+  async removeBooksFromRefresh(bookIds: Array<string> = []) {
+    if (!bookIds.length) return;
+
+    console.log(`Readwise Official plugin: removing book ids ${bookIds.join(', ')} from refresh list`);
+    this.settings.booksToRefresh = this.settings.booksToRefresh.filter(n => !bookIds.includes(n));
+    await this.saveSettings();
+  }
+
+  async reimportFile(vault: Vault, fileName: string) {
     try {
-      fetch(
-        `${baseURL}/api/refresh_book_export`,
-        {
-          headers: {...this.getAuthHeaders(), 'Content-Type': 'application/json'},
-          method: "POST",
-          body: JSON.stringify({exportTarget: 'obsidian', books: [bookId]})
-        }
-      ).then(response => {
-        if (response && response.ok) {
-          let booksToRefresh = this.settings.booksToRefresh;
-          this.settings.booksToRefresh = booksToRefresh.filter(n => ![bookId].includes(n));
-          this.saveSettings();
-          vault.delete(vault.getAbstractFileByPath(fileName));
-          this.startSync();
-        } else {
-          this.notice("Failed to reimport. Please try again", true);
-        }
-      });
+      this.notice("Deleting and reimporting file...", true);
+      await vault.delete(vault.getAbstractFileByPath(fileName));
+      const bookId = this.settings.booksIDsMap[fileName];
+      await this.refreshBookExport([bookId]);
     } catch (e) {
       console.log("Readwise Official plugin: fetch failed in Reimport current file: ", e);
     }
   }
 
-  startSync() {
+  async startSync() {
     if (this.settings.isSyncing) {
       this.notice("Readwise sync already in progress", true);
     } else {
       this.settings.isSyncing = true;
-      this.saveSettings();
-      this.requestArchive();
+      await this.saveSettings();
+      await this.requestArchive();
     }
     console.log("Readwise Official plugin: started sync");
   }
 
   async onload() {
-    // @ts-ignore
+    await this.loadSettings();
+
+    // @ts-expect-error - no type for isMobile
     if (!this.app.isMobile) {
       this.statusBar = new StatusBar(this.addStatusBarItem());
       this.registerInterval(
         window.setInterval(() => this.statusBar.display(), 1000)
       );
     }
-    await this.loadSettings();
-    this.refreshBookExport = debounce(
-      this.refreshBookExport.bind(this),
-      800,
-      true
-    );
 
-    this.refreshBookExport(this.settings.booksToRefresh);
     this.app.vault.on("delete", async (file) => {
-      const bookId = this.settings.booksIDsMap[file.path];
-      if (bookId) {
+      // user has "Resync deleted files" enabled
+      if (this.settings.refreshBooks) {
+        const bookId = this.settings.booksIDsMap[file.path];
+        if (!bookId) return;
+
+        // queue the book for refresh
         await this.addBookToRefresh(bookId);
       }
-      this.refreshBookExport();
-      delete this.settings.booksIDsMap[file.path];
-      this.saveSettings();
     });

```

Comment:
i'm not convinced that this `on(delete)` listener is needed at all with my other changes.

- entries are no longer removed from `booksIDsMap`
- `refreshBookExport` will now look for the delta between `booksIDsMap` and your actual system
- ... and if `refreshBooks` (re-sync deleted files) is enabled, will sync those missing files

Reply:
i deleted the `on(delete)` listener and verified that it is now unnecessary by running through the test set in OP